hearthstone/simulator/core/cards.py

```diff
@@ -309,10 +309,11 @@ def __init__(self, cards: List[MonsterCard]):
         for card in cards:
             self.cards_by_tier[card.tier].add(card)
 
-    def draw(self, player: 'Player', num: int) -> List['MonsterCard']:
+    def draw(self, player: 'Player', num: int, predicate: Optional[Callable] = lambda card: True) -> List['MonsterCard']:
```

Comment:
I think this should take a "max_tier" as an argument instead of a player.  The max_tier argument can even be optional.

Reply:
We need `player` for other attributes as well here.